51 todo setting up todo list canvas

ChooChooChao opened the pull request:
Objectives
- Set up the HTML page for the Todo List canvas

Acceptance Criteria
- [x ] Set up the 3 columns with a few dummy tasks hard coded
- [x ] Set up page title
- [x ] Include the dashboard headers with navigation to other pages

shjucsd committed:
setup basic files for todo

ChooChooChao committed:
ToDo List Canvas

I set up the very initial Todo list canvas. Created three separate columns: "Not Started," "In Progress," "Done." Placed a add new task button with no functionality. Included dummy tasks as place holders to be replaced with the functionality of adding/modifying tasks.

shjucsd reviewed (changes requested):
The css looks good. For the hard-coded data, can we have the number of tasks display for in progress and done as well just so that we are consistent with the ui? Similarly, can you add the tags for different labels like Work, Personal.

ChooChooChao committed:
Todo Canvas Update

Adjusted the todo list canvas to match more of the UI design. Added the number of tasks indicator next to the title of each column. Added the task categories including "work," "people," "school." Included a delete button.

Cloud1379 committed:
Update to-do-app.js

Cloud1379 committed:
Update to-do.html

alt012ucsd committed:
update

shjucsd committed:
working on localStorage for todo

shjucsd committed:
Merge branch '51-todo-setting-up-todo-list-canvas' of https://github.com/cse110-sp24-group30/cse110-sp24-group30 into 51-todo-setting-up-todo-list-canvas

shjucsd committed:
got basic localStorage for add function

shjucsd committed:
localStorage works for delete and updates label accordingly

shjucsd committed:
ran prettier for consistent formatting

shjucsd committed:
dynamically updating the display count for each column according to
localStorage

shjucsd committed:
Merge branch 'main' into 51-todo-setting-up-todo-list-canvas

alt012ucsd reviewed (approved):
Everything looks good, all base functionality accounted for.

shjucsd reviewed (approved):
I added the functionality for updating the localStorage on adding and deleting the tasks and dynamically updating the display count as well, i also updated the formatting using prettier

adironene commented on to_do/to-do-app.js:
```diff
@@ -0,0 +1,267 @@
+  };
+
+  if (existing) {
+    console.log("existing");
```
if we are logging stuff in console, I would make it more detailed. Like existing what? + more context

adironene commented on to_do/to-do.html:
```diff
@@ -0,0 +1,105 @@
+                                <p id="work-category">work</p>
+                            </div>
+                            <span id="due-date">Due Date: 5/28</span>
+                            <!-- <div class="buttons">
```
remove unused comments

adironene commented on to_do/to-do-app.js:
```diff
@@ -0,0 +1,267 @@
+  updateCount();
+}
+
+function addTask(toDoContainer, existing, category) {
```
I would add a bit more comments throughout the doc to tell the reader what the code is doing. It's self-explanatory after reading through a few times but having comments before each "group" of tasks would be very helpful!

adironene reviewed (changes requested):
Amazing work! Good job dividing the tasks up in different functions. Just left a few comments for suggestions and good practice.

Also **todo** - please attach a video for UI demo

kabirsachdev7 reviewed (approved):
Foundation for the backend is good. Definitely meets the acceptance criteria. Phoebe touched on some points already, but just want to say to add more comments pls wherever you can as you continue building the todo.

shjucsd commented:
for the logging into the console comment, i think we just had those to check our implementation before and that the code was entering the correct places, we can remove them and i can add comments for all the functions so that they can be used to create jsdocs

shjucsd committed:
added comments to functions, removed unuseful comments and log
statements

adironene reviewed (approved):
Fixes look good!
